### PR TITLE
[PowerPC] Fix assert exposed by PR 95931 in LowerBITCAST

### PR DESCRIPTION
**Pull request by syzaara**
Hit Assertion failed: Num < NumOperands && "Invalid child # of SDNode!" 
Fix by checking opcode and value type before calling getOperand.

### CODE REVIEW

**Review by lei137 — APPROVED** [review]
LGTM
Thx!

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp [hunk 1]
```diff
@@ -9470,12 +9470,13 @@ SDValue PPCTargetLowering::LowerBITCAST(SDValue Op, SelectionDAG &DAG) const {
...
+    return SDValue();
+
   SDValue Lo = Op0.getOperand(0);
   SDValue Hi = Op0.getOperand(1);
-
-  if ((Op.getValueType() != MVT::f128) ||
-      (Op0.getOpcode() != ISD::BUILD_PAIR) || (Lo.getValueType() != MVT::i64) ||
-      (Hi.getValueType() != MVT::i64) || !Subtarget.isPPC64())
+  if ((Lo.getValueType() != MVT::i64) || (Hi.getValueType() != MVT::i64) ||
+      !Subtarget.isPPC64())
```
Comment (lei137) [inline llvm/lib/Target/PowerPC/PPCISelLowering.cpp:9479]: nit: maybe good to move `!Subtarget.isPPC64()` to begining of cond on line 9473.

File: llvm/test/CodeGen/PowerPC/f128-bitcast.ll [hunk 2]
```diff
@@ -86,3 +86,25 @@ entry:
   ret i64 %1
 }
 
+define <4 x i32> @test(i512 %a) {
```
Comment (amy-kwan) [inline llvm/test/CodeGen/PowerPC/f128-bitcast.ll:89]: nit: Can we maybe be a bit more descriptive for the test title? Not sure if something like `truncBitcast` makes sense, or something like that?

**Review by amy-kwan — APPROVED** [review]
LGTM.

**Comment by tgross35** [comment]
This should probably get backported since https://github.com/llvm/llvm-project/pull/95931 was in https://github.com/llvm/llvm-project/pull/95931

**Comment by syzaara** [comment]
> This should probably get backported since #95931 was in #95931 (or I guess that one reverted, since I said in the backport that it seemed low risk but that was apparently erroneous :/)

I see 95931 change is still in release/19.x so it would be good to have this PR back ported as well.

**Comment by syzaara** [comment]
/cherry-pick https://github.com/llvm/llvm-project/commit/22067a8eb43a7194e65913b47a9c724fde3ed68f

**Pull request merged**